opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciAskForPin.java

```diff
@@ -21,7 +23,11 @@
 
     @Override
     protected void doRealExecution(DelegateExecution execution, HbciContext context) {
-        redirectExecutor.redirect(execution, context, redir -> redir.getParameters().getProvidePsuPassword());
+        redirectExecutor.redirect(execution, context, redir -> {
+            HbciProtocolConfiguration.UrlSet urlSet = ProtocolAction.SINGLE_PAYMENT.equals(context.getAction())
```

Comment:
Suggest to move
```
HbciProtocolConfiguration.UrlSet urlSet = ProtocolAction.SINGLE_PAYMENT.equals(context.getAction())
                    ? redir.getPis() : redir.getAis();
```
directly into HbciContext:
```
class HbciContext {

   @JsonIgnore
    public UrlSet getActiveUrlSet(HbciProtocolConfiguration config) {
       return ProtocolAction.SINGLE_PAYMENT.equals(context.getAction())
                    ? redir.getPis() : redir.getAis();
    }
}
```

because it is already used twice - here and in HbciAskForTan